graylog2-web-interface/src/components/extractors/extractors_configuration/GrokExtractorConfiguration.jsx

```diff
@@ -19,8 +21,34 @@ class GrokExtractorConfiguration extends React.Component {
     onExtractorPreviewLoad: PropTypes.func.isRequired,
   };
 
+  static defaultProps = {
+    exampleMessage: undefined,
+  };
+
   state = {
     trying: false,
+    patterns: [],
+  };
+
+  componentDidMount() {
+    this.loadData();
+  }
+
+  componentWillUnmount() {
+    if (this.loadPromise) {
+      this.loadPromise.cancel();
+    }
+  }
+
+  loadData = () => {
+    this.loadPromise = GrokPatternsStore.loadPatterns((patterns) => {
+      if (!this.loadPromise.isCancelled()) {
```

Comment:
If a user clicks on this page but leaves the page before the `loadPatterns` is returned. The resolving promise will make an error in the console, since he tries to setState on an not existing component.

With `isCancelled` I can cancel the promise when unmounting the component.